packages/core/src/amazonqTest/error.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { ToolkitError } from '../shared/errors'
+
+export const techinalErrorCustomerFacingMessage =
+    'I am experiencing technical difficulties at the moment. Please try again in a few minutes.'
+const defaultTestGenErrorMessage = 'Amazon Q encountered an error while generating tests. Try again later.'
+export class TestGenError extends ToolkitError {
+    constructor(
+        error: string,
+        code: string,
+        public statusCode: string,
+        public customerFacingMessage: string
```

Comment:
Name it `uiMessage`. This is really a special case, and will lead to confusion and hard-to-follow code, and inconsistent UX. Why can't customers see the actual error message? This will cause drift and confusion, and adds steps to troubleshooting.

Reply:
>Why can't customers see the actual error message

Unfortunately error messages received from the backend is not aligning to the UX messages. So IDE need to handle this explicitly.
I will update `customerFacingMessage ` to `uiMessage `